src/clj/ataru/applications/application_store.clj

```diff
@@ -291,14 +291,26 @@
        first
        :virkailija_oid))
 
+(def chars-to-filter [\u0000])
+
+(defn filter-chars [coll chars]
+  (apply str (remove #((set chars) %) coll)))
+
+(defn filter-answer-values [value]
+  (if (vector? value)
+    (into [] (map filter-answer-values value))
+    (filter-chars value chars-to-filter)))
+
 (defn add-application [new-application applied-hakukohteet form session]
     (jdbc/with-db-transaction [conn {:datasource (db/get-datasource :db)}]
       (info (str "Inserting new application"))
-      (let [selection-id   (:selection-id new-application)
-            virkailija-oid (when-let [secret (:virkailija-secret new-application)]
+      (let [clean-answers (vec (map #(util/map-kv % filter-answer-values) (:answers new-application)))
```

Comment:
`(mapv #(update % :value filter-answer-values) (:answers new-application))` pitäis hoitaa sama asia hiukan selkeämmin.